query/query/sockets.py

```diff
@@ -0,0 +1,265 @@
+import json
+import time
+import os
+import socket
+import struct
+import logging
+from hailtop.utils import sync_retry_transient_errors, TransientError
+
+
+log = logging.getLogger('query.sockets')
+
+
+class ServiceBackendJavaConnector:
+    def __init__(self):
+        self.fname = '/sock/sock'
+        while not os.path.exists(self.fname):
+            time.sleep(1)
+
+    def connect(self) -> 'ServiceBackendSocketSession':
+        sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        sync_retry_transient_errors(sock.connect, self.fname)
+        return ServiceBackendSocketSession(
+            ServiceBackendSocketAPI(sock))
+
+
+class ServiceBackendSocketSession:
```

Comment:
I feel like you don't need the connector or the session, just the API (which I'd call a connection?)  It can connect to the socket on connect, and make existence of the socket file a transient error?

Reply:
I don't grok your comment about transient errors, but I made the rest of the changes in b81b8fae32